pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotInstanceRestletResource.java

```diff
@@ -0,0 +1,204 @@
+    final PinotResourceManagerResponse resp = pinotHelixResourceManager.addInstance(instance);
+    if (resp.status == PinotResourceManagerResponse.ResponseStatus.failure) {
+      throw new WebApplicationException("Instance exists already",
+          javax.ws.rs.core.Response.Status.CONFLICT);
+    }
+    return new SuccessResponse("Instance successfully created");
+  }
+
+  // TODO: @consumes text/plain but swagger doc says json. Does that work. It's better this way if it works
+  @POST
+  @Path("/instances/{instanceName}/state")
+  @Produces(MediaType.APPLICATION_JSON)
+  @Consumes(MediaType.TEXT_PLAIN)
+  @ApiOperation(value = "Create a new instance", consumes = MediaType.APPLICATION_JSON, notes = "Creates a new instance with given instance config")
+  @ApiResponses(value = {@ApiResponse(code = 200, message="Instance successfully created"),
+      @ApiResponse(code=409, message="Instance exists already"),
+      @ApiResponse(code = 400, message="Bad Request"),
+      @ApiResponse(code = 500, message = "Internal error")})
+  public SuccessResponse addInstance(
+      @ApiParam(value = "Instance name", required = true,
+          example = "Server_a.b.com_20000 | Broker_my.broker.com_30000")
+      @PathParam("instanceName") String instanceName,
+      String state
+  ) {
+    // TODO: state should be moved to path or form parameter
+    if (! pinotHelixResourceManager.instanceExists(instanceName)) {
+      throw new WebApplicationException("Instance " + instanceName + " does not exist",
+          Response.Status.NOT_FOUND);
+    }
+    PinotResourceManagerResponse response;
+    if (StateType.ENABLE.name().equalsIgnoreCase(state)) {
+      response = pinotHelixResourceManager.enableInstance(instanceName);
+    } else if (StateType.DISABLE.name().equalsIgnoreCase(state)) {
+      response = pinotHelixResourceManager.disableInstance(instanceName);
+    } else if (StateType.DROP.name().equalsIgnoreCase(state)) {
+      response = pinotHelixResourceManager.dropInstance(instanceName);
+    } else {
+     throw new WebApplicationException("Unknown state " + state + " for instance request",
+         Response.Status.BAD_REQUEST);
+    }
+    if (response.isSuccessful()) {
+      return new SuccessResponse("Request to " + state + " instance " + instanceName  + " is successful");
+    }
+    throw new WebApplicationException("Failed to " + state + " instance " + instanceName,
+        Response.Status.INTERNAL_SERVER_ERROR);
+  }
+
+  @DELETE
+  @Path("/instances/{instanceName}")
+  @Consumes(MediaType.TEXT_PLAIN)
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Delete an instance", consumes = MediaType.APPLICATION_JSON,
+      notes = "Deletes an instance of given name")
+  @ApiResponses(value = {@ApiResponse(code = 200, message="Instance successfully deleted"),
+      @ApiResponse(code = 404, message = "Instance not found"),
+      @ApiResponse(code = 409, message = "Instance can not be deleted"),
```

Comment:
Can we put the previous error message back? Might be more clear
@Response(statusCode = "409", description = "Forbidden operation typically because the instance is live or "
          + "idealstates still contain some information of this instance \n")